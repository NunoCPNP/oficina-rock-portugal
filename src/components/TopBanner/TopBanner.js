import useTranslation from 'next-translate/useTranslation'

import { Container } from './TopBanner.styles'

const TopBanner = () => {
  const { t } = useTranslation('common')

  return <Container>TOP BANNER</Container>
}

export default TopBanner
