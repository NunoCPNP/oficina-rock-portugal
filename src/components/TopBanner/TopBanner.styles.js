import mq from '@/styles/mediaQueries'
import styled from '@emotion/styled'

export const Container = styled.div`
  margin: 4rem;
  background-color: var(--darkLiver);
  padding: 0.6rem;
  border-radius: 0.8rem;

  ${mq('s')} {
    margin: 2rem;
  }

  span {
    text-decoration: underline;
  }

  h4 {
    text-align: center;
    text-transform: uppercase;
    color: var(--snow);
  }
`
