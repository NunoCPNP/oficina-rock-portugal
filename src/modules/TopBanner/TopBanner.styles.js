import mq from '@/styles/mediaQueries'
import styled from '@emotion/styled'

export const Container = styled.div`
  margin: 4rem;
  background-color: var(--ghostWhite);
  padding: 0.6rem;
  border-radius: 0.5rem;
  box-shadow: 4px 4px 5px 2px rgba(0, 0, 0, 0.1);

  ${mq('s')} {
    margin: 2rem;
  }

  span {
    text-decoration: underline;
  }

  h4 {
    text-align: center;
    text-transform: uppercase;
    color: var(--darkLiver);
  }
`
