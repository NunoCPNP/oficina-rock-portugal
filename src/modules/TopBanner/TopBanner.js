import { Container } from './TopBanner.styles'
import Trans from 'next-translate/Trans'

import { useSettingsState } from '@/hooks/useSettings'

const TopBanner = () => {
  const { topBanner } = useSettingsState()

  if (!topBanner) {
    return null
  }

  return (
    <Container>
      <Trans
        i18nKey="common:free-shipping-banner"
        components={{
          a: <h4 />,
          b: <span />,
        }}
      />
    </Container>
  )
}

export default TopBanner
