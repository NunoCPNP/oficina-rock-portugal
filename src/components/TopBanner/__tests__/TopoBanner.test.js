import { render } from '@testing-library/react'

import TopBanner from '../TopBanner'

const mockContext = jest.fn()

jest.mock('@/hooks/useSettings', () => ({
  ...jest.requireActual('@/hooks/useSettings'),
  useSettingsState: () => mockContext(),
}))

describe('<TopBanner /> spec', () => {
  it('Should match snapshot', () => {
    mockContext.mockReturnValueOnce({ topBanner: true })

    const { container } = render(<TopBanner />)

    expect(container).toMatchSnapshot()
  })

  it('Should render translation keys', () => {
    mockContext.mockReturnValueOnce({ topBanner: true })

    const { getByText } = render(<TopBanner />)

    expect(getByText('common:free-shipping-banner')).toBeInTheDocument()
  })

  it('Should not render if topBanner is false', () => {
    mockContext.mockReturnValueOnce({ topBanner: false })

    const { container } = render(<TopBanner />)

    expect(container).toMatchSnapshot()
  })
})
