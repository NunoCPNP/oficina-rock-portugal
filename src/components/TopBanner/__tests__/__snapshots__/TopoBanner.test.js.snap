// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBanner /> spec Should match snapshot 1`] = `
.emotion-0 {
  margin: 4rem;
  background-color: var(--darkLiver);
  padding: 0.6rem;
  border-radius: 0.8rem;
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    margin: 2rem;
  }
}

.emotion-0 span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0 h4 {
  text-align: center;
  text-transform: uppercase;
  color: var(--snow);
}

<div>
  <div
    class="emotion-0"
  >
    common:free-shipping-banner
  </div>
</div>
`;

exports[`<TopBanner /> spec Should not render if topBanner is false 1`] = `<div />`;
